libsolidity/codegen/ReturnInfoCollector.h

```diff
@@ -0,0 +1,84 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Component that translates Solidity code into Yul at statement level and below.
+ */
+#pragma once
+
+#include <liblangutil/EVMVersion.h>
+#include <libsolidity/ast/Types.h>
+
+#include <functional>
+#include <string>
+
+namespace solidity::frontend
+{
+
+class FunctionCall;
+
+/// Holds a set of properties of an external function call's return values.
+struct ReturnInfo
```

Comment:
I wonder whether this whole thing should move to the AST or the type system instead...
It's not that much related to code generation besides the returndataVariable, but that's just "passed through" anyways...
I'd imagine we could even re-use parts of the logic for old codegen (if we wanted to).

Reply:
It is already used in the old codegen.

Reply:
Ah ok, haven't seen that yet, I'll have another look.

Reply:
Ah, ok, my comment was before https://github.com/ethereum/solidity/pull/8246/commits/ef2fec631310ed1004ef38f4754c27b766f7f788 - after that it's much nicer, yes!
And one can argue that it's a codegen thing and actually belongs here and not into the type system, since it depends on EVM version. So alright!